enterprise/internal/batches/store/changesets.go

```diff
@@ -1467,3 +1471,18 @@ func uiPublicationStateColumn(c *btypes.Changeset) *string {
 	}
 	return uiPublicationState
 }
+
+// CleanDetachedChangesets deletes changesets that have been detached after duration specified.
+func (s *Store) CleanDetachedChangesets(ctx context.Context, retention time.Duration) (err error) {
+	ctx, _, endObservation := s.operations.cleanDetachedChangesets.With(ctx, &err, observation.Args{LogFields: []log.Field{
+		log.String("Retention", retention.String()),
+	}})
+	defer endObservation(1, observation.Args{})
+
+	return s.Exec(ctx, sqlf.Sprintf(cleanDetachedChangesetsFmtstr, retention/time.Second))
+}
+
+const cleanDetachedChangesetsFmtstr = `
+-- source: enterprise/internal/batches/store/changesets.go:CleanDetachedChangesets
+DELETE FROM changesets WHERE detached_at < (NOW() - (%s * interval '1 second'));
```

Comment:
We usually use `s.now()` so we can mock the `now()` date in tests.

```suggestion
DELETE FROM changesets WHERE detached_at < (%s - (%s * interval '1 second'));
```